[Doc] Update README.md (#3461)
LMCache/LMCache: README.md

<div align="center">
  <p align="center">

>     <img src="asset/logo.png" alt="lmcache logo">

Reviewer: Can we apply this?

</details>

## About

LMCache is a **KV cache management layer** for LLM inference. It turns KV cache from a temporary state into reusable *AI-native knowledge* that can be *stored* persistently, *reused* across multiple serving engines, *monitored* with observability stack, and *transformed* for better generation quality. As a result, LMCache **reduces TTFT** (time-to-first-token) and **improves throughput**, especially for long-context agentic, multi-turn conversation, and knowledge-augmented workloads (e.g., RAG).

LMCache is **vendor-neutral**. It can be used as a KV cache layer for a range of mainstream open-source serving engines, inference frameworks, hardware vendors, storage systems, and infrastructure providers. The vendor neutrality allows users to freely switch between serving engines and storage vendors, while reusing the stored KV caches. 

<p align="center">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="asset/deployment_modes_dark.png">
  <source media="(prefers-color-scheme: light)" srcset="asset/deployment_modes_light.png">
  <img alt="LMCache Deployment Modes" src="asset/deployment_modes_light.png">
</picture>
</p>

### Key features

> - **Engine-independent deployment**: LMCache, as a standalone damon process, manages KV cache independently from the inference engine process, so that KV cache will not be lost even if the inference engine crashes (i.e., no fate-sharing with engines).

Reviewer: damon -> daemon

Author: Thank you for spotting those! @DongDongJu. I’ve applied your suggested fix, so it should be good now. Thanks again!

- **Persistent, tiered KV cache offloading and reuse**: Move KV caches out of GPU memory into a tiered storage hierarchy spanning CPU memory, local storage, and remote backends, enabling reuse across requests, sessions, and engine instances to reduce repeated prefill computation and improve TTFT.

- **Production-level KV cache observability**: LMCache provides a rich set of KV cache observability metrics, including typical Kubernetes metrics (health monitoring, performance diagnostics), KV-cache-specific metrics (request-level and token-level prefix cache hits, lifecycle, request-level KV cache performance), management metrics (user-specific usage) and more.

> - **Pluggable storage and transport backends**: Easily integrate remote storage and KV transfer backends through a unified interface, enabling KV cache offloading and sharing across storage providers. Through the interface, LMCache support storage backend including CPU RAM, local disk (SSD), Redis/Valkey, Mooncake, InfiniStore, S3-compatible object storage, NIXL, and GDS.

Reviewer: Through the -> Through this?

- **Non-prefix KV reuse**: Extend KV reuse beyond prefix caching by reusing cached KV blocks for repeated text chunks at any position in prompt. This leverages CacheBlend to selectively recompute tokens for quality recovery.

- **PD disaggregation and KV transfer**: Support KV cache transfer from prefill workers to decode workers over NVLink, RDMA, or TCP through transport layers such as NIXL.

> - **Pluggable KV transformation**: Simple interface for for researchers to write compression, token dropping, and custom serialization through a flexible SERDE interface. 

Reviewer: for for -> for

- [2025/09] NVIDIA Dynamo integrates LMCache, accelerating LLM inference ([blog](https://blog.lmcache.ai/en/2025/09/18/nvidia-dynamo-integrates-lmcache-accelerating-llm-inference/)).
- [2025/08] 🎉 LMCache hits 5,000+ GitHub stars ([blog](https://blog.lmcache.ai/en/2025/08/28/%f0%9f%8e%89-lmcache-hits-5000-github-stars-thank-you-community/)).
- [2025/08] LMCache supports gpt-oss (20B/120B) on day 1 ([blog](https://blog.lmcache.ai/en/2025/08/05/lmcache-supports-gpt-oss-20b-120b-on-day-1/)).
- [2025/07] Get faster LLM inference and cheaper responses with LMCache and Redis ([Redis blog](https://redis.io/blog/get-faster-llm-inference-and-cheaper-responses-with-lmcache-and-redis/)).
- [2025/07] LMCache extends its turbo-boost to multimodal models in vLLM V1 ([blog](https://blog.lmcache.ai/en/2025/07/03/lmcache-extends-its-turbo-boost-to-multimodal-models-in-vllm-v1/)).
- [2025/06] LLM Production Stack goes cross-hardware: AMD, Arm and Ascend ([blog](https://blog.lmcache.ai/en/2025/06/20/llm-production-stack-goes-cross-hardware-ascend-arm-and-amd-support-incoming/)).

</details>

## About

> LMCache is a **KV cache management layer** for LLM inference. It turns KV cache from a temporary state into reusable *AI-native knowledge* that can be *stored* persistently, *reused* across multiple serving engines, and *transformed* for better generation quality. As a result, LMCache **reduces TTFT** (time-to-first-token) and **improves throughput**, especially for long-context agentic, multi-turn conversation, and knowledge-augmented workloads (e.g., RAG).

Reviewer 2: ```suggestion
LMCache is a **KV cache management layer** for LLM inference. It turns KV cache from a temporary state into reusable *AI-native knowledge* that can be *stored* persistently, *reused* across multiple serving engines, *monitored* with observability stack and *transformed* for better generation quality. As a result, LMCache **reduces TTFT** (time-to-first-token) and **improves throughput**, especially for long-context agentic, multi-turn conversation, and knowledge-augmented workloads (e.g., RAG).
```

<!-- LMCache supports two deployment modes:

- **Multi-process (MP) mode** *(recommended)*: LMCache, as a standalone damon process, manages KV cache independently from the inference engine process, and the inference engines connect to LMCache through connectors over ZMQ. In MP mode, a single LMCache server can connect with multiple engine instances, share cache across them, and expose unified management and observability endpoints. 

- **In-process mode**: LMCache runs inside the inference engine process through connectors this mode is limited by python GILs. -->

<p align="center">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="asset/deployment_modes_dark.png">
  <source media="(prefers-color-scheme: light)" srcset="asset/deployment_modes_light.png">
  <img alt="LMCache Deployment Modes" src="asset/deployment_modes_light.png">
</picture>
</p>

### Key features

> - **Engine-independent deployment**: LMCache, as a standalone damon process, manages KV cache independently from the inference engine process, and the inference engines connect to LMCache through connectors over ZMQ. A single LMCache server can connect with multiple engine instances, share cache across them, and expose unified management and observability endpoints. KV cache will not be lost even if the inference engine crashes (i.e., no fate-sharing with engines). 

Reviewer 2: ```suggestion
- **Engine-independent deployment**: LMCache, as a standalone damon process, manages KV cache independently from the inference engine process, so that KV cache will not be lost even if the inference engine crashes (i.e., no fate-sharing with engines).
```

> - **Persistent, tiered KV cache offloading and reuse**: Move KV caches out of GPU memory into a tiered storage hierarchy spanning CPU memory, local storage, and remote backends, enabling reuse across requests, sessions, and engine instances to reduce repeated prefill computation and improve TTFT.

Reviewer 2: ```suggestion
- **Prompt caching all over the cluster**: By enabling KV cache offloading and reuse using tiered storage hierarchy spanning CPU memory, local storage, and remote backends, LMCache allows you to perform prompt caching all over the cluster and significantly reduces repeated prefill computation and improve TTFT.
```

> 

Reviewer 2: ```suggestion
Production-level KV cache observability support: LMCache provides KV cache observability metrics, ranging from typical k8s metrics (health monitoring, performance diagnostics), KV-cache-specific metrics (request-level and token-level prefix cache hits, lifecycle, request-level KV cache performance), management metrics (user-specific usage) and more.
```

> - **Pluggable storage and transport backends**: Easily integrate remote storage and KV transfer backends through a unified interface, enabling KV cache offloading and sharing across storage providers. Through the interface, LMCache support storage backend including CPU RAM, local disk (SSD), Redis/Valkey, Mooncake, InfiniStore, S3-compatible object storage, NIXL, and GDS.

Reviewer 2: Widest storage and transport solution coverage
